pinecone/client.go

```diff
@@ -37,6 +38,21 @@ type NewClientBaseParams struct {
 	SourceTag  string
 }
 
+// NewIndexConnParams holds the parameters for creating an IndexConnection to a Pinecone index.
+//
+// Fields:
+//   - Host: The host URL of the Pinecone index. This is required. To find your host url use the DescribeIndex or ListIndexes methods.
+//     Alternatively, the host is displayed in the Pinecone web console.
+//   - Namespace: Optional index namespace to use for operations. If not provided, the default namespace of "" will be used.
+//   - AdditionalMetdata: Optional additional metdata to be sent with each RPC request.
```

Comment:
Nit: it might be worth giving an example of what metadata one would send with an RPC request